secv_guis/bimask_app/main_window.py

```diff
@@ -32,95 +32,15 @@
-    This function assumes that the given ``elts`` have been sampled from an
-    exponential distribution. Then inferes lambda, using the unbiased ML
-    estimator (see https://en.wikipedia.org/wiki/Exponential_distribution)
-    and returns the threshold ``t`` that fulfills ``keep_p_value`` for the
-    distribution above ``t``.
-    """
-    assert 0 < keep_p_value <= 1, "area_p_conserve must be in (0, 1]"
-    threshold = -np.log(keep_p_value) / lmbd
-    return threshold
-
-
-def pmap_to_mask(pmap, keep_highest_pval=0.05, discard_lowest_pval=0.5):
-    print(np.max(pmap),np.min(pmap),keep_highest_pval)
-    return (pmap/(2**16-1))>=keep_highest_pval
-    """
-    This method performs the following steps:
-
-    1. Assuming that pmap values are exponentially distributed, extracts the
-         unbiased lambda parameter and the cumulative distribution.
-    2. Applies threshold to the given low/high p-values
-
-    This method uses the standard connected component extraction mechanism
-    in Python, i.e. ``scipy.ndimage.measurements.label`` and
-    ``skimage.measure.regionprops``.
-
-    :param pmap: A float array of shape ``h, w``.
-    :param keep_highest_pval: The p-value designing the amount of top 'pmap'
-      scores to be surely kept.
-    :param discard_lowest_pval: The p-value designing the amount of bottom
-      'pmap' scores to be surely discarded.
-    :returns: The output mask
-    """
-    # print("PMAP TO MASK CALLED:", keep_highest_pval, discard_lowest_pval)
-    assert keep_highest_pval < discard_lowest_pval, \
-        "highest-P-val must be lower than lowest-P-val."
-
-    lmbd = exp_lambda_estimator(pmap.ravel())
-    t_keep_above = exp_threshold(keep_highest_pval, lmbd)
-    t_discard_below = exp_threshold(discard_lowest_pval, lmbd)
-    out_mask = apply_hysteresis_threshold(pmap, t_discard_below, t_keep_above)
-
-    # sure keep: pmap>=t_keep_above   sure discard: pmap <= t_discard_below
-
-    # t = exp_threshold(keep_below_pval, pmap.ravel())
-    # mask = pmap >= t
-    # # label performs connected component identification, by default
-    # # using a cross-shape structuring elt.
-    # labels, nlabels = label(mask)
-    # # compute a set of feats for each isolated component
-    # props = regionprops(labels, pmap)
-    # #
-    # areas = [p.area for p in props]
-    # areas_threshold = exp_threshold(keep_p_value, areas)
-    #
-    # out_mask = np.zeros_like(mask)
-    # for p in props:
-    #     if p.area >= areas_threshold:
-    #         coords_y, coords_x = p.coords.T
-    #         out_mask[(coords_y, coords_x)] = True
-
-    return out_mask
-
+def pmap_to_mask(pmap,upper_percentile,lower_percentile,percentile_max=100):
+    pmap=np.array(pmap)
+    values = np.sort(pmap[::-1].flatten())
+    up = int((len(values)-1) * upper_percentile /percentile_max)
+    lp= int((len(values)-1) * lower_percentile / percentile_max )
+    pmap[pmap>values[up]]=0
+    pmap[pmap<values[lp]]=0
+    pmap=pmap>0
+    return pmap
```

Comment:
Really nice that you got rid of the cancer above! I'll just assume this works.
But please document the new function with a docstring explaining:

* what is expected for each input parameter in terms of meaning and datatype
* what does the function do/return
* Any further notes/warnings/comments...

And please check the linting: `=` in assignments should have whitespaces before and after. This is true for almost every operator (the exception is the `=` in the parameter list, and in this case it is permitted to have a compact `<` `>` inside the array indexes. But in general don't try to save space with that, it is really worth keeping the consistency and adding a newline if you get above 80 characters. flake8 :)